go.mod

```diff
@@ -13,6 +13,7 @@ require (
 	github.com/pkg/errors v0.9.1
 	github.com/spf13/cobra v1.1.1
 	github.com/spf13/viper v1.7.0
+	github.com/stretchr/testify v1.6.1
```

Comment:
for nested structures, JSONs, YAMLs, this seems to be the best library possible; other similar-ish lightweight libraries are [is](https://pkg.go.dev/github.com/matryer/is#section-readme) and a cool library which we might like better [gocmp](github.com/google/go-cmp/cmp).

# Reasons for preferring go-cmp for comparing structs

1. [Cleaner one line-ish diffs of structs](https://blog.artwolf.in/a?ID=2e0cd2a0-3dfd-4dbc-a87d-c40647552b9a)
2. [Mentioned by Dave Cheney in testing blogs](https://dave.cheney.net/tag/testing)